doc/dev_guide/writing_extensions.rst

```diff
@@ -85,61 +78,31 @@ The flowchart below can help you decide where to add
 a new data analysis function. Notice that only if no suitable package exists
 for your function, you should consider creating your own.
 
-..  This is the original mermaid code. It produces a nicer looking diagram
-    with the defaults, but, as of version 0.3.1, it raises an exception in
-    ReadTheDocs, so we use graphviz below instead.
-
-    .. mermaid::
-
-       graph TD
-
-         A(New function needed!)
-         B{Is it useful for data of any type and dimensions?}
-         C(Contribute it to BaseSignal)
-         D{Does a SignalxD for the required dimension exist in HyperSpy?}
-         E[Contribute new SignalxD to HyperSpy]
-         F{Is the function useful for a specific type of data only?}
-         G(Contribute it to SignalxD)
-         H{Does a signal for that sort of data exists?}
-         I(Contribute to package providing the relevant signal)
-         J(Create you own package and signal subclass to host the funtion)
-         A-->B
-         B-- Yes -->C
-         B-- No  -->D
-         D-- Yes -->F
-         D-- No  -->E
-         E-->F
-         F-- Yes -->H
-         F-- No  -->G
-         H-- Yes -->I
-         H-- No -->J
-
-
-.. graphviz::
-
-    digraph G {
-         A [label="New function needed!"]
-         B [label="Is it useful for data of any type and dimensions?",shape="diamond"]
-         C [label="Contribute it to BaseSignal"]
-         D [label="Does a SignalxD for the required dimension exist in HyperSpy?",shape="diamond"]
-         E [label="Contribute new SignalxD to HyperSpy"]
-         F [label="Is the function useful for a specific type of data only?",shape="diamond"]
-         G [label="Contribute it to SignalxD"]
-         H [label="Does a signal for that sort of data exist?",shape="diamond"]
-         I [label="Contribute to package providing the relevant signal"]
-         J [label="Create you own package and signal subclass to host the funtion"]
-         A->B
-         B->C [label="Yes"]
-         B->D [label="No"]
-         D->F [label="Yes"]
-         D->E [label="No"]
-         E->F
-         F->H [label="Yes"]
-         F->G [label="No"]
-         H->I [label="Yes"]
-         H->J [label="No"]
-
-    }
+
+.. mermaid::
+
```

Comment:
For a theme that works well in dark mode add:
```
%%{init: {'theme':'redox'}}%%
```
(alternative would be the `'dark'` theme.

Reply:
For a live example see:
https://mermaid.live/edit#pako:eNpdjl1vgyAUhv8KOVdbokYFqyVpk1rvd7FdbeyCCH6kFQzFtJvxvw9Jui47V7zvcx5ghloLCRTCMGSq1qrpW8oU8mM7OUiKjBT6xpTfaM76WnfcWPRW3dcQOjx9MHi1rmbw-fynR7vdHpUzg0rW_aXXisHywCUKwz06OvdltA6ig9P_4-qBS48hgNb0Aqg1kwxgkGbga4R5VRn4XzOg7ii4OTFganHOyNW71sNdM3pqO6ANP19cmkbBrax63ho-_LZGKiHNUU_KAiUY-0uAznBbY7RJybYgOUnSJN6SAL6AYpxEuCBJlqU5JjjPlgC-_atxVORZ7CbBKd7kcZwtP7LobTA